uber/templates/registration/checkin.html

```diff
@@ -0,0 +1,120 @@
+<script type="text/javascript">
+    $(function() {
+        $('.dialog-form').dialog({ autoOpen: false });
+
+        $('.attendee-checkin').button().on('click', function () {
+            var attendee_id = $(this).attr('data-attendeeid');
+            var dialog = $('#' + attendee_id).dialog({
+                autoOpen: false,
+                height: 450,
+                width: 500,
+                modal: true,
+                buttons: [{
+                    text: 'Save & Check In',
+                    click: function() {
+                        check_in(attendee_id);
+                        dialog.dialog('close');
+                    }
+                }, {
+                    text: 'Cancel',
+                    click: function () {
+                        dialog.dialog('close');
+                    }
+                }],
+                // FIXME: OMG TERRIBLE POSITION HAX
+                open: function () {
+                  $('.ui-dialog').css('position', 'fixed')
+                    .css('top', '5%')
+                    .css('left', '33%');
+                }
+            });
+            dialog.find('input.num').keypress(function (event) {
+                if (event.keyCode === 13) {  // Enter key
+                    event.preventDefault();
+                    dialog.dialog('close');
+                }
+            });
+            dialog.dialog('open');
+        });
+    });
+
+    var check_in = function (id) {
+        if ($('#age_' + id).val() == {{ AGE_UNKNOWN }}) {
+            alert('You may not check someone in without confirming their age.');
+            return;
+        }
+        $.post('check_in', {
+            id: id,
+            csrf_token: csrf_token,
+            age_group:  $('#age_' + id).val(),
+            badge_num:  $('#num_' + id).val(),
+            group: $('#group_' + id).val() || ''
+        }, function (json) {
+            var message = json.message;
+            if (json.success) {
+                message += ' &nbsp; <a href="#" onClick="undoCheckin(\'' + id + '\', ' + json.pre_badge + ') ; return false">Undo</a>';
+                $('#paid_' + id).html(json.paid);
+                $('#cin_' + id).html(json.checked_in);
+                $('#age_' + id).parent().html(json.age_group);
+                $('#num_' + id).parent().html(json.badge);
+            }
+            showTop(message);
+            if (json.increment) {
+                $('#checkin_count').html(1 + parseInt($("#checkin_count").html()));
+            }
+        }, 'json');
+    };
+    var undoCheckin = function (id, pre_badge) {
+        $.post('undo_checkin', {id: id, csrf_token: csrf_token, pre_badge: pre_badge}, function(s) {
+            var sep = location.href.indexOf('?') === -1 ? '?' : '&';
+            location.href += sep + 'message=' + encodeURIComponent(s);
+        });
+    };
+</script>
```

Comment:
It would probably be more efficient to construct this dynamically, but the page loads relatively quickly, so this should be fine for this year.